schema_based_on_diagram.sql

```diff
@@ -0,0 +1,68 @@
+CREATE TABLE patients (
+    id SERIAL,
+    name VARCHAR(100),
+    date_of_birth DATE NOT NULL,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+    id SERIAL,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status VARCHAR(50),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE treatments (
+    id SERIAL,
+    type VARCHAR(50),
+    name VARCHAR(100),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE invoice_items (
+    id SERIAL,
+    unit_price decimal,
+    quantity INT,
+    total_price decimal,
+    invoice_id INT,
+    treatment_id INT,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE invoices (
+    id SERIAL,
+    total_amount decimal,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT,
+    PRIMARY KEY (id)
+);
+
+ALTER TABLE medical_histories
+ADD CONSTRAINT medical_fk FOREIGN KEY (patient_id) REFERENCES patients (id);
+
+ALTER TABLE invoice_items
+ADD CONSTRAINT invoice_fk FOREIGN KEY (invoice_id) REFERENCES invoices (id);
+
+ALTER TABLE invoice_items
+ADD CONSTRAINT treatment_fk FOREIGN KEY (treatment_id) REFERENCES treatments (id);
+
+ALTER TABLE invoices
+ADD CONSTRAINT medical_invoice_fk FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id);
+
+CREATE TABLE medical_history_treatments (
+    id SERIAL PRIMARY KEY,
+    medical_history_id INT,
+    treatment_id INT,
+    CONSTRAINT medical_history_fk
+    FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+    CONSTRAINT treatment_fk
+    FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+
+SELECT * FROM medical_histories
+INNER JOIN medical_history_treatments
+ON medical_histories.id = medical_history_treatments.medical_history_id
+INNER JOIN treatments
+ON medical_history_treatments.treatments_id = treatments.id;
```

Comment:
- [ ] <b>REQUIRED:</b> You haven't created all indexes the project requires you to create for all FOREIGN KEYs. Kindly do that.

Reply:
Hello @rukundoeric,
First of all, thanks a lot for the comment and for the review.
But to understand well what more is required from us,  please bring more clarification.
We created all the tables from the diagram.
After that, we linked them together altering the tables thru the foreign keys, as the diagram shows - all of them.
And at the end, in the case of many to many relations between medical_histories and treatments tables, we created an extra table.

Reply:
![diagram](https://user-images.githubusercontent.com/83596135/150012929-ff932e0b-ee83-4ae1-9b06-0b029863729b.png)

This is the diagram that our code produces - created on [dbdiagram.io](https://dbdiagram.io/d/61e6e6e6bb7a646986f8b1aa)